### PR TITLE
[FEATURE] Cache_fetch_user

### LINKED ISSUE #8: Fetch and cache User data
## Task
- [x] Implement the api service for the issue
- [x] Implement the cache system.

## Acceptance criteria
- [x] should validate already saved user offline

### PR DESCRIPTION
**Pull request by alvinmarshall**
local database to save user object
some added items:
* UserDao
* LocalUserImpl
* UserEntity
* UserMapper

resolves: #8

### CODE REVIEW

File: local-source/dependencies.gradle [hunk 1]
```diff
@@ -0,0 +1,64 @@
+ext {
+
+    Rx = [
+            rxKotlin: "io.reactivex.rxjava3:rxkotlin:${RxKotlin}",
+            rxJava  : "io.reactivex.rxjava3:rxjava:${RxJava}",
+            bridge  : "com.github.akarnokd:rxjava3-bridge:3.0.0"
```
Comment (alvinmarshall) [inline local-source/dependencies.gradle]: forgot to extract version number

**Review by call-me-3v — APPROVED** [review]
Awesome :). Good job done

**Review by call-me-3v — APPROVED** [review]
I have reviewed the codes and everything is going on well as expected.

**Review by call-me-3v — APPROVED** [review]
I have reviewed the codes and everything is going on well as expected.

**Pull request merged**